nav2_route/src/node_spatial_tree.cpp

```diff
@@ -65,7 +70,10 @@ bool NodeSpatialTree::findNearestGraphNodeToPose(
     return false;
   }
 
-  node_id = ret_index[0];
+  for (int i = 0; i < num_of_nearest_nodes_ && i < static_cast<int>(ret_index.size()); ++i) {
+    node_ids.push_back(ret_index[i]);
```

Comment:
You mentioned a potential ordering issue - is it possible that these aren't promised to be in order?

Reply:
I am not quite sure which ordering issue you are referring too :).

Reply:
Something in slack about the order of ... something not being as you expected. Maybe my memory is failing me

Reply:
I think that was in reference to the order of nodes found by the breath first search algorithm. :). At any rate, it would be good to double check that the order is deterministic from the rtree.